src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy

```diff
@@ -0,0 +1,266 @@
+import de.dkfz.roddy.tools.LoggerWrapper
+import de.dkfz.roddy.tools.RoddyConversionHelperMethods
+import de.dkfz.roddy.tools.TimeUnit
+import groovy.transform.CompileStatic
+import groovy.util.slurpersupport.GPathResult
+import org.xml.sax.SAXParseException
+
+import java.time.Duration
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.util.regex.Matcher
+
+@CompileStatic
+abstract class GridEngineBasedJobManager<C extends Command> extends ClusterJobManager<C> {
+
+    private static final LoggerWrapper logger = LoggerWrapper.getLogger(GridEngineBasedJobManager)
+
+    public static final String WITH_DELIMITER = '(?=(%1$s))'
+
+    GridEngineBasedJobManager(BEExecutionService executionService, JobManagerOptions parms) {
+        super(executionService, parms)
+    }
+
+    @Override
+    boolean getDefaultForHoldJobsEnabled() { return true }
+
+    @Override
+    String getSubmissionCommand() {
+        return "qsub"
+    }
+
+    protected int getPositionOfJobID() {
+        return 0
+    }
+
+    protected int getPositionOfJobState() {
+        return 4
+    }
+
+    @Override
+    protected Map<BEJobID, JobState> queryJobStates(List<BEJobID> jobIDs) {
+        StringBuilder queryCommand = new StringBuilder(getQueryJobStatesCommand())
+
+        if (jobIDs && jobIDs.size() < 10) {
+            queryCommand << " " << jobIDs*.id.join(" ")
+        }
+
+        if (isTrackingOfUserJobsEnabled)
+            queryCommand << " -u $userIDForQueries "
+
+        ExecutionResult er = executionService.execute(queryCommand.toString())
+        List<String> resultLines = er.resultLines
+
+        Map<BEJobID, JobState> result = [:]
+
+        if (!er.successful) {
+            if (strictMode) // Do not pull this into the outer if! The else branch needs to be executed if er.successful is true
+                throw new BEException("The execution of ${queryCommand} failed.\n\t" + er.resultLines?.join("\n\t").toString())
+        } else {
+            if (resultLines.size() > 2) {
+
+                for (String line : resultLines) {
+                    line = line.trim()
+                    if (line.length() == 0) continue
+                    if (!RoddyConversionHelperMethods.isInteger(line.substring(0, 1)))
+                        continue //Filter out lines which have been missed which do not start with a number.
+
+                    //TODO Put to a common class, is used multiple times.
+                    line = line.replaceAll("\\s+", " ").trim()       //Replace multi white space with single whitespace
+                    String[] split = line.split(" ")
+                    final int ID = getPositionOfJobID()
+                    final int JOBSTATE = getPositionOfJobState()
+                    if (logger.isVerbosityMedium())
+                        logger.postAlwaysInfo(["QStat BEJob line: " + line,
+                                               "	Entry in arr[" + ID + "]: " + split[ID],
```

Comment:
Is there a guarantee that the array has the appropriate size? If not, add a check/assertion and throw a useful error message. Here and also for the other fields.

Reply:
WontFix: I think this is removed in the current development branch.